uk.co.compucorp.civicrm.hrcore/backstop_data/page-objects/ssp-leave-absences-my-leave-report.js

```diff
@@ -60,6 +60,63 @@ module.exports = (function () {
           resolve(this.waitForModal('ssp-leave-request', '.chr_leave-request-modal__form'));
         }.bind(this));
       }.bind(this));
+    }*/
+    /**
+     * User clicks on the edit/respond action
+     * @param {Number} row number corresponding to leave request in the list
+     * @return {Promise}
+     */
+    editRequest: function (row) {
+      var casper = this.casper;
+
+      casper.then(function () {
+        casper.click('body > ul.dropdown-menu:nth-of-type(' + (row || 1) + ') li:first-child a');
+        // As there are multiple spinners it takes more time to load up
+        casper.waitWhileVisible('.modal-content .spinner:nth-child(1)');
+        casper.waitWhileVisible('leave-request-popup-details-tab .spinner');
+      });
+
+      return this;
+    },
+
+    openCommentsTab: function () {
+      var casper = this.casper;
+      var selector = '.chr_leave-request-modal__tab .uib-tab:nth-of-type(2) a';
+
+      casper.waitForSelector(selector)
+        .then(function () {
+          casper.click(selector);
+        });
+
+      return this;
+    },
+
+    /**
+     * Triggers an error alert
+     * by selecting a date that does not correspond to any Absence Period
+     *
+     * @return {Promise}
+     */
+    triggerErrorAlert: function () {
+      var casper = this.casper;
+      var modalSelector = '.chr_leave-request-modal';
+      var datepickerSelector = modalSelector + ' .uib-datepicker-popup';
+      var previousMonthButtonSelector = datepickerSelector + ' .fa-chevron-left';
+
+      casper.then(function () {
```

Comment:
can you try removing the `.then`'s and see if it works?

Reply:
Yes, I spent a lot of time on that. It does not work without `then`, it does not wait for a selector.